vcs/src/main/java/org/openjdk/skara/vcs/hg/HgRepository.java

```diff
@@ -599,21 +599,23 @@ public boolean isAncestor(Hash ancestor, Hash descendant) throws IOException {
 
     @Override
     public void rebase(Hash hash, String committerName, String committerEmail) throws IOException {
-        var current = currentBranch().name();
+        var current = currentBranch().orElseThrow(() ->
+                new IOException("No current branch to rebase upon")
```

Comment:
Should this mention the detached head then as well perhaps?

Reply:
Scratch that, guess this is hg :)

Reply:
Yes, this is in `HgRepository`, so a "detached HEAD" is not possible. I actually don't think `currenBranch` _can_ fail for `HgRepository`, since the current branch is part of the commit metadata in Mercurial (not just a ref as in git). I added a message anyhow to distinguish this scenario.